modules/auxiliary/gather/ldap_query.rb

```diff
@@ -213,7 +310,39 @@ def normalize_entries(entries)
       entry = entry.to_h
       entry_keys = entry.keys
       entry_keys.each do |key|
-        entry[key] = entry[key].map { |v| Rex::Text.to_hex_ascii(v) }
+        if key.to_s.match(/guid$/i)
+          # Get the the entry[key] object will be an array containing a single string entry,
+          # so reach in and extract that string, which will contain binary data.
+          binguid = entry[key][0]
+          decoded_guid = decode_guid_binstring(binguid)
+          entry[key][0] = decoded_guid
+        elsif key == :cacertificate
+          raw_key_data = entry[key][0]
+          _certificate_file, read_data = read_der_certificate_file(raw_key_data)
+          entry[key][0] = read_data
+        elsif key == :objectsid
+          # Advice taken from https://ldapwiki.com/wiki/ObjectSID
+          object_sid_raw = entry[key][0]
+          sid_string = decode_binary_data_to_sid(object_sid_raw)
+          entry[key][0] = sid_string
+        # This one is hard as there are fields that end with the string Time that are technically
+        # enumerations or strings themselves, but these sometimes only exist on older versions
+        # of Windows Server such as 2012.
+        elsif key == :creationtime
+          timestamp = entry[key][0]
+          time_string = convert_nt_timestamp_to_time_string(timestamp)
+          entry[key][0] = time_string
+        elsif key == :systemflags
+          flags = entry[key][0]
+          converted_flags_string = convert_system_flags_to_string(flags)
+          entry[key][0] = converted_flags_string
+        elsif key.to_s.match(/pwdage$/) || key.to_s.match(/lockoutduration$/i)
+          timestamp = entry[key][0]
+          time_string = convert_pwd_age_to_time_string(timestamp)
+          entry[key][0] = time_string
+        else
+          entry[key] = entry[key].map { |v| Rex::Text.to_hex_ascii(v) }
```

Comment:
I haven't verified this against the new PR, but I assume the issue still exists:

Looks like we'll want to remove this for genuine UTF-8 characters, this was the output I got for an account created with the name `wîth.dìáçriticš`


Metasploit:
```
msf6 auxiliary(gather/ldap_query) > run rhost=192.168.123.13 username=Administrator@adf3.local password=p4$$w0rd

... etc ...

CN=w\xc3\xaeth.d\xc3\xac\xc3\xa1\xc3\xa7ritic\xc5\xa1 CN=Users DC=adf3 DC=local
===============================================================================

 Name                Attributes
 ----                ----------
 displayname         w\xc3\xaeth.d\xc3\xac\xc3\xa1\xc3\xa7ritic\xc5\xa1
 name                w\xc3\xaeth.d\xc3\xac\xc3\xa1\xc3\xa7ritic\xc5\xa1
 samaccountname      w\xc3\xaeth.d\xc3\xac\xc3\xa1\xc3\xa7ritic\xc5\xa1
 useraccountcontrol  66048
 userprincipalname   w\xc3\xaeth.d\xc3\xac\xc3\xa1\xc3\xa7ritic\xc5\xa1@adf3.local
```

Expected:

```
displayname       wîth.dìáçriticš
... etc...
userprincipalname wîth.dìáçriticš@adf3.local
```

Similar to other tools:

<img width="782" alt="image" src="https://user-images.githubusercontent.com/60357436/188766593-fd72b94d-ed75-4bae-b5f0-8d300701fd16.png">

Although it looks like ldapsearch doesn't handle this scenario intuitively either; The internet suggests to pipe to perl to have it handle the characters correctly for its base64 decoding https://access.redhat.com/solutions/43621 🤷 

```
$ ldapsearch -H ldap://dc3.adf3.local -D "Administrator@adf3.local" -w 'p4$$w0rd' -b 'dc=adf3,dc=local' '(|(objectClass=inetOrgPerson)(objectClass=user)(sAMAccountType=805306368)(objectClass=
posixAccount)(objectClass=GsAccount)(objectClass=GsSIPUser))' 'displayName'

... etc ...

# w\C3\AEth.d\C3\AC\C3\A1\C3\A7ritic\C5\A1, Users, adf3.local
dn:: Q049d8OudGguZMOsw6HDp3JpdGljxaEsQ049VXNlcnMsREM9YWRmMyxEQz1sb2NhbA==
displayName:: d8OudGguZMOsw6HDp3JpdGljxaE=
```

Reply:
I thought we were doing this precisely because the table output couldn't handle UTF-8 characters?

Reply:
Rex-table supports valid utf-8 characters:

```
>> table = Rex::Text::Table.new('SortIndex' => -1, 'Indent' => 4, 'Header' => 'Example', 'Columns' => ['key', 'value'], 'Rows' => [['a', 'b'], ['sam
', "w\xC3\xAEth.d\xC3\xAC\xC3\xA1\xC3\xA7ritic\xC5\xA1".force_encoding("utf-8")], ["foo", "\xf0\x9f\x8e\x89\xf0\x9f\x8e\x89\xf0\x9f\x8e\x89".force_encoding('utf-8')]]); puts table.to_s
Example
=======                                                                                                                                             

    key  value
    ---  -----
    a    b
    sam  wîth.dìáçriticš
    foo  🎉🎉🎉
```

<img width="1158" alt="image" src="https://user-images.githubusercontent.com/60357436/188920587-bd295d12-9939-440a-9724-540793b6f6ea.png">

> the table output couldn't handle UTF-8 characters?

iirc Your original issue was attempting to output arbitrary binary data as though it were valid text? 👀

Reply:
Hmm guess I misinterpreted that case then, I'll try see if I can get this updated. Thanks for reporting 👍

Reply:
@adfoster-r7 Would you be okay if we put this into another PR? I think this may be a larger change. I do think its still valid but determining how we would determine if something is within the UTF-8 range is a bit harder of a challenge given different fonts would have different levels of UTF-8 character support which might only amplify this problem.

Reply:
The current implementation isn't exactly wrong, it's just escaping non ASCII values.

According to https://ldap.com/ldapv3-wire-protocol-reference-asn1-ber/ though in the "Octet String Values" section, strings are "usually the bytes that comprise the UTF-8 representation of that text". Knowing that, it might make sense to treat the string as UTF-8 encoded (pretty sure it's ASCII-8bit right now) and if there are any bad characters, revert back to this instance because it's probably just some other opaque blob. We just jump straight to a hex dump, but we'd be missing any embedded text data which you can see now (without these changes) in the case of the certificate.